lib/charms/sdcore_nms_k8s/v0/fiveg_core_gnb.py

```diff
@@ -0,0 +1,348 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Library for the `fiveg_core_gnb` relation.
+
+This library contains the Requires and Provides classes for handling the `fiveg_core_gnb`
+interface.
+
+The purpose of this library is to provide a way for a 5G Core to provide network information and
+configuration to CUs/gNodeBs.
+
+To get started using the library, you need to fetch the library using `charmcraft`.
+
+```shell
+cd some-charm
+charmcraft fetch-lib charms.sdcore_nms_k8s.v0.fiveg_core_gnb
+```
+
+Add the following libraries to the charm's `requirements.txt` file:
+- pydantic
+- pytest-interface-tester
+
+Charms providing the `fiveg_core_gnb` relation should use `FivegCoreGnbProvides`.
+The class `PLMNConfig` represents the configuration of a PLMN for the CU/gNodeB. It is composed by
+the Mobile Country Code (MCC), the Mobile Network Code (MNC), the Slice Service Type (SST) and the
+Slice Differentiator (SD). Each CU can be configured with a single Tracking Area Code (TAC) and
+multiple PLMNS.
+
+Typical usage of this class would look something like:
+
+    ```python
+    ...
+    from charms.sdcore_gnbsim_k8s.v0.fiveg_core_gnb import FivegCoreGnbProvides, PLMNConfig
+    ...
+
+    class SomeProviderCharm(CharmBase):
+
+        def __init__(self, *args):
+            ...
+            self.fiveg_core_gnb_provider = FivegCoreGnbProvides(
+                charm=self,
+                relation_name="fiveg_core_gnb"
+                )
+            ...
+            self.framework.observe(
+                self.on.fiveg_core_gnb_provider_relation_changed,
```

Comment:
```suggestion
                self.on.fiveg_core_gnb_relation_changed,
```

Reply:
Actually I think we should be running this logic on the event related to network config change (in our case, it's gonna be the Pebble notification). There's no point in running this on `relation_changed`, which would be triggered when gNB pushes it's name to the data bag; at that time, there won't be any configuration available for that gNB.